src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1235,11 +1262,37 @@ CHIP_ERROR CASESession::SendSigma3()
     return err;
 }
 
-CHIP_ERROR CASESession::HandleSigma3(System::PacketBufferHandle && msg)
+struct CASESession::Sigma3Work
+{
+    // Status of background processing.
+    CHIP_ERROR status;
+
+    // Session to use after background processing.
+    CASESession * session;
+
+    // Sequence number used for coordination.
```

Comment:
I think this probably deserves a longer comment explaining how this sequence number works.

Reply:
I added a slightly longer comment here.